PopPUNK/unwords.py

```diff
@@ -0,0 +1,35 @@
+import json
+import gzip
+import random
+import string
+import pkg_resources
+
+# Based on a simple interpretation of https://simple.wikipedia.org/wiki/Syllable
+def gen_unword(unique=True):
+    # Download from https://github.com/dwyl/english-words/raw/master/words_dictionary.json
+    word_list = pkg_resources.resource_stream(__name__, 'data/words_dictionary.json.gz')
+    with gzip.open(word_list, 'rb') as word_list:
+        real_words = json.load(word_list)
+
+    vowels = ["a", "e", "i", "o", "u"]
+    trouble = ["q", "x", "y"]
+    consonants = set(string.ascii_lowercase) - set(vowels) - set(trouble)
+
```

Comment:
I wonder if it might be worth supplementing consonants with phonemes, e.g. https://www.boardman.k12.oh.us/userfiles/363/Phonological%20Awareness/44Phonemes.pdf. There are some extra vowel sounds there too (e.g. "oo"). Might be worth only including some as prefix (e.g. "wh") and some as suffix (e.g. "h").

Reply:
Tried the phonemes but looked a bit strange, I think it would probably work better to sample them in frequency proportional to how much they're actually used, and add more rules about order. I think I'll keep it simple for now